salt/runners/digicertapi.py

```diff
@@ -533,28 +525,30 @@ def gen_key(minion_id, dns_name=None, password=None, key_len=2048):
 
         salt-run digicert.gen_key <minion_id> [dns_name] [password]
     """
-    keygen_type = "RSA"
 
-    if keygen_type == "RSA":
-        if HAS_M2:
-            gen = RSA.gen_key(key_len, 65537)
-            private_key = gen.as_pem(
-                cipher="des_ede3_cbc", callback=lambda x: bytes(password)
-            )
-        else:
-            gen = RSA.generate(bits=key_len)
-            private_key = gen.exportKey("PEM", password)
-        if dns_name is not None:
-            bank = "digicert/domains"
-            cache = salt.cache.Cache(__opts__, syspaths.CACHE_DIR)
-            try:
-                data = cache.fetch(bank, dns_name)
-                data["private_key"] = private_key
-                data["minion_id"] = minion_id
-            except TypeError:
-                data = {"private_key": private_key, "minion_id": minion_id}
-            cache.store(bank, dns_name, data)
-    return private_key
+    private_key = rsa.generate_private_key(public_exponent=65537, key_size=key_len)
+    encryption = (
+        serialization.BestAvailableEncryption(bytes(password, encoding="utf-8"))
```

Comment:
This has a high likely-hood of breaking backwards comparability with older versions of the module which still use m2crypto. Though I am not sure how important that is for this runner.

Reply:
While it's true that the `PyCA/cryptography` code uses a different format and encryption for the generated RSA keys, the keys are only ever used (from what I could see) to sign Certificate Signing Requests.
This is done through the `openssl req -new -key $PKEY_PATH ...` command, which supports both the old and new key formats.
So old versions of the module should still be able to work with the keys generated by the new version, since the keys are only used indirectly through that `openssl` command.

Reply:
Okay, this makes sense.